src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -399,13 +400,16 @@ private static function remove_json_comments( $array ) {
 	 *                                     so it only fetches published posts.
 	 * @return array Custom Post Type for the user's origin config.
 	 */
-	public static function get_user_data_from_wp_global_styles( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
-		if ( ! $theme instanceof WP_Theme ) {
-			$theme = wp_get_theme();
+	public static function get_user_data_from_wp_global_styles( $theme_or_stylesheet, $create_post = false, $post_status_filter = array( 'publish' ) ) {
```

Comment:
IMO we should either name this `$stylesheet` or `$theme_dir`, both of which are used in certain places in core.

`$theme_or_stylesheet` is a bit too verbose, and looking at the function, this method should have really just received the stylesheet when it was introduced in the first place. So I think it's fine to name the parameter based on that. We can simply keep support for passing a `WP_Theme` like you have here, but let's encourage folks to pass a stylesheet throughout the code.